capstone/src/test/java/com/google/sps/data/DataStoreTest.java

```diff
@@ -34,14 +41,99 @@ private void doTest() {
     assertEquals(2, ds.prepare(new Query("test")).countEntities(withLimit(10)));
   }
 
-  /* Commented these out since tests fail, will add more tests in future PRS */
-  //   @Test
-  //   public void testInsert1() {
-  //     doTest();
-  //   }
+  @Test
+  public void testStateleakInsert1() {
+    doTest();
+  }
+
+  @Test
+  public void testStateleakInsert2() {
+    doTest();
+  }
+
+  public boolean inLocalStore(Entity entity) {
+    try {
+      datastore.get(entity.getKey());
+    } catch (EntityNotFoundException e) {
+      return false;
+    }
+
+    return true;
+  }
+
+  /**
+   * Creates an entity based on given survey response values
+   *
+   * @param kind Namespace in datastore
+   * @param zipCode Location of where survey was taken
+   * @param id Unique id given to each survey
+   * @param score Sentiment score for the survey response
+   * @return Entity data store object representing a survey response
+   */
+  public Entity getMockEntity(String kind, String zipCode, int id, float score) {
+    Entity entity = new Entity(kind, id);
+    entity.setProperty("zipCode", zipCode);
+    entity.setProperty("score", score);
 
-  //   @Test
-  //   public void testInsert2() {
-  //     doTest();
-  //   }
+    return entity;
+  }
+
+  @Test
+  public void testMockEntityStorage() throws IOException {
+    ArrayList<Entity> testEntities = new ArrayList<Entity>();
+    Entity a = getMockEntity("ResponseTest", "94101", 1, 0.5f);
+    Entity b = getMockEntity("ResponseTest", "94151", 2, 0.83f);
+
+    testEntities.add(a);
+    testEntities.add(b);
+
+    datastore.put(testEntities);
+
+    Entity entity = new Entity("ResponseTest", 2);
+
+    assertTrue(inLocalStore(entity));
+  }
+
+  /**
+   * Creates entities based on real survey file through the ReadData class
+   *
+   * @param entities Arraylist of entities that will be modified with file data
+   * @param file The file that will be parsed to create entities
+   * @return void
+   */
+  public void getEntities(ArrayList<Entity> entities, File file) throws IOException {
+    ReadData readData = new ReadData();
+    readData.entitiesFromFile(entities, file);
+  }
+
+  @Test
+  public void testInsertEntity() throws IOException {
```

Comment:
Seems like you just have a few normal csvs (which seems like typical data). What are some edge cases that you can think of? Some of the ones that come to mind are empty files and adding the same file twice

Reply:
Fortunately, the surveys api ensures that the survey responses are complete when downloaded through the script. Duplicate files are written as one file by the script as well

Reply:
Its really cool that the surveys api does a lot of this for you. However, I think it would still be nice to show that your system is robust and can deal with these situations (which from glancing over your code it can).